### ISSUE TITLE
Style/Attr not catching `attr` usage

### ISSUE THREAD

**Issue by ngan** [rubocop/rubocop#11812]
## Expected behavior

Given a ruby file:

```ruby
class Foo
  attr :foo

  def bar
    puts 'ok'
  end
end
```

```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

I would have expected the `Style/Attr` cop to catch the `attr` usage--but it doesn't.
If I were to remove the `#bar` method, the cop then catches the offense.

```ruby
class Foo
  attr :foo
end
```
```
Inspecting 1 file
C

Offenses:

ngan.rb:2:3: C: [Correctable] Style/Attr: Do not use attr. Use attr_reader instead.
  attr :foo
  ^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

## RuboCop version

```
1.49.0 (using Parser 3.2.2.0, rubocop-ast 1.28.0, running on ruby 3.1.4) [arm64-darwin22]
  - rubocop-graphql 0.13.0
  - rubocop-performance 1.16.0
  - rubocop-rails 2.16.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.19.0
  - rubocop-sorbet 0.6.7
```
